draft-ietf-oauth-selective-disclosure-jwt.md

```diff
@@ -580,6 +580,78 @@ To ensure readability and replicability, the examples in this specification do
 not contain decoy digests unless explicitly stated. For an example
 with decoy digests, see (#example-simple_structured).
 
+### Recursive Disclosures {#recursive_disclosures}
+
+The algorithms above are compatible with "recursive disclosures", in which one
+selectively disclosed field reveals the existence of more selectively
+disclosable fields.  For example, consider the following JSON structure:
+
+```json
+{
+    "family_name": "Möbius",
+    "nationalities": ["DE", "FR", "UK"]
+}
+```
+
+When the Holder has multiple nationalities, the issuer may wish to allow the holder to reveal
+each of those nationalities individually.
+This can be accomplished by first making the "FR" and "UK" entries within
+the array selectively disclosable, and then the whole "nationalities" field.
+
+``` ascii-art
+{
+    "family_name": "Möbius",
+    "nationalities": [
+        "DE",
+        { "...": "Yr1KUNYXI6WzJKhP4Hwyu9Q1ejubVCcJRG1C9FiWr+c" },
```

Comment:
I can't seem to reproduce these hash values, which makes me a wee bit nervous. But the full Disclosures aren't shown so it's a bit of a stretch anyway. But I'm going to plow ahead and hope I'm not doing something wrong thereby assuming @bifurcation made some mistake or just made the values up.  Blasphemous, I know, but I gotta move forward.

Reply:
wait... "+" isn't in the base64url alphabet! WTF?

Reply:
Reversing hashes is hard! 

But "+" was the clue I needed and extraneous padding on the not shown Disclosures was the key. Now @bifurcation owes me something in Vancouver. Beer is customary item but I'm 14 years sober so he might get away with something cheaper.

Reply:
> Happy to supply whatever compensatory item makes sense! 

We will need to make something happen in YVR. 

>Thanks for picking this up, I was planning to handle it on a flight last Friday, but then it didn't have in-flight wifi.

I figured something came up. Meanwhile I'm trying to get something out before I-D cutoff and have limited co-author availability due to vacation etc. 

> Any errors here are totally on me. I was creating the hashes manually, typing / copy / pasting into `python3` REPL, and translating the results back manually to the URL safe alphabet.

It's hard to imagine how any mistakes made it into that pipeline. But there were a few...